### DISCUSSION TITLE
Struggling to install codon-jit on Linux

### DISCUSSION THREAD

**Discussion by SpaceAzur** [exaloop/codon#378] [Q&A]
Hi,

I'm struggling installing codon-jit on Ubuntu 22.04.2

I have successfully installed codon 
`/bin/bash -c "$(curl -fsSL https://exaloop.io/install.sh)"`

I have correctly update my PATH variable `export PATH=/home/administrateur/.codon/bin:$PATH` in my .bashrc
```
codon --version
0.16.0
```
As I want to install codon-jit, this is the error I get
```
➜  ~ sudo pip3 install codon-jit                              
Collecting codon-jit
  Using cached codon-jit-0.1.4.tar.gz (5.0 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [1 lines of output]
      Cannot find Codon. Please either install Codon (/bin/bash -c "$(curl -fsSL https://exaloop.io/install.sh)"), or set CODON_DIR if Codon is not in PATH or installed in ~/.codon
      [end of output]
```

As I red the doc, I shall need to set my CODON_PYTHON variable as well. The doc give an example on macOS.
```
export CODON_PYTHON=/path/to/libpython.X.Y.so
/usr/local/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/lib/libpython3.9.dylib
```

What is the equivalent for Linux. So many .so files in /usr/lib/python3. Which one is to be use to set the CODON_PYTHON variable. Is someone can help ?

**Comment by kostya** [comment]
in my docker file I use this
```
RUN apt install -y libpython3-dev
RUN /bin/bash -c "$(curl -fsSL https://exaloop.io/install.sh)"
ENV CODON_PYTHON=/usr/lib/x86_64-linux-gnu/libpython3.10.so
```

**Comment by inumanag** [comment]
Can you try doing `CODON_PATH=$HOME/.codon pip install codon-jit`?

Installing with `sudo` (not recommended, btw) probably breaks it because codon is not installed as root.

**Comment by athishrao** [comment]
To find the `libpythonX.Y.so` that need to be set to the env variable `CODON_PYTHON` in Linux, you can run this small python script to find out:

```
import sysconfig
py_so_path = "/".join(sysconfig.get_config_vars('LIBDIR', 'INSTSONAME'))
print(py_so_path)
```
